Userland/Services/WindowServer/ClientConnection.cpp

```diff
@@ -1115,4 +1115,22 @@ void ClientConnection::set_flash_flush(bool enabled)
     Compositor::the().set_flash_flush(enabled);
 }
 
+void ClientConnection::set_window_parent_from_id(i32 client_id, i32 parent_id, i32 child_id)
+{
+    auto child_window = window_from_id(child_id);
+    if (!child_window)
+        did_misbehave("SetWindowParentFromId: Bad child window ID");
+
+    auto client_connection = from_client_id(client_id);
+    if (!client_connection)
+        did_misbehave("SetWindowParentFromId: Bad client ID");
+
+    auto parent_window = client_connection->window_from_id(parent_id);
+    if (!parent_window)
+        did_misbehave("SetWindowParentFromId: Bad parent window ID");
+
+    child_window->set_parent_window(*parent_window);
+    child_window->set_rect(parent_window->rect());
```

Comment:
The `set_rect` makes this not very generic, but it is necessary to make the prompts appear in the centre for the following commits.

Can we set the size elsewhere or are we forced into something strange here?

Reply:
See the `get_window_rect_from_client` commit.